### ISSUE TITLE
Hybrid Search returns documents with "vector_distance": 2.0 ignoring distance_threshold

### ISSUE THREAD

**Issue by ganvoa** [typesense/typesense#1294]
## Description

Hybrid Search returns documents with `"vector_distance": 2.0` when using `filter_by` on certain fields, ignoring `distance_threshold` option in `vector_query`

## Steps to reproduce

<details>
<summary>This is my request</summary>

```JSON
{
    "searches": [
        {
            "collection": "products_v2",
            "exclude_fields": "embedding,clickCount,delta,discount,img,priceUpdatedAt,category,bp",
            "query_by": "brand,name,embedding",
            "limit": 24,
            "offset": 0,
            "q": "NINTENDO JUEGO NINTENDO SWITCH MORTAL KOMBAT 1",
            "vector_query": "embedding:([], distance_threshold:0.35)",
            "filter_by": "timestamp:>1696906182859"
        }
    ]
}
```
</details>

This works as expected, returning only 8 documents with vector_distance <= 0.35

[Response with 8 documents](https://gist.github.com/ganvoa/846dc0b1f39a31ced0e8b1f4fa5808e3#file-response1-json) (too long to put here)

This response only contains 2 documents where `"retailId": "paris"`. If i add that condition to the `filter_by` the  problem occurs.

<details>
<summary>Request with filter_by including retailId filter</summary>

```JSON
{
    "searches": [
        {
            "collection": "products_v2",
            "exclude_fields": "embedding,clickCount,delta,discount,img,priceUpdatedAt,category,bp",
            "query_by": "brand,name,embedding",
            "limit": 24,
            "offset": 0,
            "q": "NINTENDO JUEGO NINTENDO SWITCH MORTAL KOMBAT 1",
            "vector_query": "embedding:([], distance_threshold:0.35)",
            "filter_by": "timestamp:>1696906182859&&retailId:=paris"
        }
    ]
}
```
</details>

Now, the response returns 23 documents, 2 included in the first response + 21 documents with `"vector_distance": 2`

[Response with 23 documents](https://gist.github.com/ganvoa/846dc0b1f39a31ced0e8b1f4fa5808e3#file-response2-json) (too long to put here)

I get the same result with every retailId but `retailId: ripley` 

## Expected Behavior

Hybrid Search should return only documents with a maximun vector_distance specified in distance_threshold

## Actual Behavior

Returns documents with `vector_distance: 2` ignoring the distance_threshold value

## Metadata

**Typesense Version**: 0.25.1

**OS**: Linux

**Comment by jasonbosco** [comment]
`distance_threshold` is only applied to results that are fetched via vector search. The results you see with a `vector_distance: 2`, are actually _keyword_ search results, which are not impacted by distance_threshold. 

We're in the process of fixing this API bug which is causing keyword search results to have a distance_threshold returned in the API response, which leads to this confusion.

**Comment by ganvoa** [comment]
Thanks! Is there any issue where can i track this bug? Couldnt find it.

**Comment by kishorenc** [comment]
This has been fixed in recent 0.25.2.rc builds.
